instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RoutesPreviewTest.kt

```diff
@@ -0,0 +1,139 @@
+        mapboxNavigation.routesUpdates()
+            .first { it.reason == RoutesExtra.ROUTES_UPDATE_REASON_NEW }
+        mapboxNavigation.routeProgressUpdates().
+            first { it.currentState == RouteProgressState.TRACKING }
+        assertEquals(RoutesPreviewUpdateReasons.PREVIEW_CLEAN_UP, currentRoutesPreview?.reason)
+        assertNull(currentRoutesPreview!!.routesPreview)
+        assertEquals(RoutesExtra.ROUTES_UPDATE_REASON_NEW, currentRoutes!!.reason)
+        assertEquals(routes, currentRoutes!!.navigationRoutes)
+        // back to free drive
+        mapboxNavigation.setNavigationRoutes(emptyList())
+        mapboxNavigation.routesUpdates()
+            .first { it.reason == RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP }
+        assertEquals(RoutesPreviewUpdateReasons.PREVIEW_CLEAN_UP, currentRoutesPreview?.reason)
+        assertNull(currentRoutesPreview!!.routesPreview)
+        assertEquals(RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP, currentRoutes!!.reason)
+        assertEquals(emptyList<NavigationRoute>(), currentRoutes!!.navigationRoutes)
+    }
+
+    @Test
+    fun route_preview_in_parallel_to_active_guidance() = sdkTest {
+        var currentRoutesPreview: RoutesPreviewUpdate? = null
+        mapboxNavigation.registerRoutesPreviewObserver { update ->
+            currentRoutesPreview = update
+        }
+        var currentRoutes: RoutesUpdatedResult? = null
+        mapboxNavigation.registerRoutesObserver { update ->
+            currentRoutes = update
+        }
+        // initial free drive
+        mapboxNavigation.startTripSession()
+        assertNull(currentRoutesPreview)
+        assertNull(currentRoutes)
+        // set routes preview
+        val initialRoutes = RoutesProvider.dc_very_short(activity).toNavigationRoutes()
+        mapboxNavigation.setRoutesPreview(initialRoutes)
+        mapboxNavigation.routesPreviewUpdates()
+            .first { it.reason == RoutesPreviewUpdateReasons.PREVIEW_NEW }
+        // start active guidance
+        mapboxNavigation.setNavigationRoutes(currentRoutesPreview!!.routesPreview!!.routesList)
+        mapboxNavigation.setRoutesPreview(emptyList())
+        mapboxNavigation.routeProgressUpdates()
+            .first { it.currentState == RouteProgressState.TRACKING }
+        // preview a different route not leaving action guidance
+        val updatedRoutes = RoutesProvider.dc_very_short_two_legs(activity).toNavigationRoutes()
+        mapboxNavigation.setRoutesPreview(updatedRoutes)
+        mapboxNavigation.routesPreviewUpdates()
+            .first { it.routesPreview?.routesList == updatedRoutes }
+        assertEquals(
+            "active guidance should track initial routes",
+            initialRoutes,
+            currentRoutes?.navigationRoutes
+        )
+        // user decided to switch to previewed routes
+        mapboxNavigation.setNavigationRoutes(currentRoutesPreview!!.routesPreview!!.routesList)
+        mapboxNavigation.setRoutesPreview(emptyList())
```

Comment:
You don't use the second API method in these tests (about changing the primary route). I think it can be tested as well.

Reply:
added tests for changing primary route